gen_docs_api.js

```diff
@@ -0,0 +1,106 @@
+const dotenv = require('dotenv')
+const env = process.env.ENV || 'DEV'
+dotenv.config({ path: '.env.' + env })
+
+// FIXME: reorder categories into a more structured order?
+// eg: wcif, events, groups, persons, cluster, ...?
+const categories = [
+  'array',
+  'boolean',
+  'cluster',
+  // FIXME: not that All useful.
+  //'display',
+  'events',
+  'groups',
+  'help',
+  'math',
+  'persons',
+  'sheets',
+  'staff',
+  'table',
+  'time',
+  'tuple',
+  'udf',
+  'util',
+  'wcif',
+]
+
+const functionToAnchorName = (name, genericParams) => {
+  if (typeof this.generated === 'undefined') {
+    this.generated = []
+  }
+  const baseName = `${name.toLowerCase()}${genericParams ? genericParams.map(t => t.toLowerCase()).join('-') : ''}`
+  let suffix = 1
+  let generatedName = baseName
+  while (this.generated.includes(generatedName)) {
+    generatedName = `${baseName}-${suffix++}`
+  }
+  this.generated.push(generatedName)
+  return generatedName
+}
+
+const functionName = (name, genericParams) => `${name}${genericParams ? `\\<${genericParams.join(', ')}>` : ''}`
+
+const stringForArg = ({ name, type, defaultValue, nullable, canBeExternal, repeated, lazy, docs }) => {
+  const properties = []
+  // FIXME: document what lazy/repeated means.
```

Comment:
I'm not sure myself:
  - `repeated` is basically `variadic` right?
  - looking at their usage, `lazy` seems to allow a partial binding for a function that will be evaluated later. eg: assuming `foo(arg: f(Round))` and `bar(e: Event, r: Round)` you could call `foo(bar(_333))`, and foo will call `arg(someround)` later, actually calling `bar(_333, someround)`. Did I understand this correctly?

Reply:
* Yes, variadic.
* The intended use case is lazy evaluation, like `Or(a, b)`, which does not need to evaluate `b` if `a` is `true` (this is useful if `b` is expensive / slow). But in practice, most use cases are what you say: an argument is actually a function, which may need to be called multiple times or using arguments that are created internally to the function. Maybe this should be split between these two cases, but I think it's working fine as a single option.

Reply:
Thanks for the clarification :)